src/main/java/wraith/fwaystones/block/WaystoneBlock.java

```diff
@@ -270,6 +269,22 @@ public ActionResult onUse(BlockState state, World world, BlockPos pos, PlayerEnt
                 blockEntity.setInventory(DefaultedList.ofSize(0, ItemStack.EMPTY));
             }
         } else {
+            // Adds a waypoint marker on a map if a player is holding one before returning with an ActionResult so
+            // that the marker is always added regardless of whether the player discovered the waystone or not.
+            MapState mapState;
+            boolean isUsingMap = false;
+            if ((mapState = FilledMapItem.getOrCreateMapState(player.getStackInHand(hand), world)) != null) {
+                isUsingMap = true;
+                boolean waystoneMarkerAdded = ((MapStateAccessor)mapState).addWaystone(world, openPos);
+                if (!Config.getInstance().discoverWaystoneOnMapUse()) {
+                    if (waystoneMarkerAdded) {
+                        return ActionResult.SUCCESS;
```

Comment:
I have tested this, and it works, but can you walk me through how the Waystone gets discovered even tho we're returning here?

Reply:
It only returns if `discoverWaystoneOnMapUse()` returns false, it's a config option.